File: Python/#844_backspaceCompare.py
class Solution:
    def backspaceCompare(self, S: str, T: str) -> bool:
        stack1,stack2 = [], []
        for i in S:
            if i =='#':
                if stack1:
                    stack1.pop()
                
            else:
                stack1.append(i)
        for i in T:
            if i =='#':
                if stack2:
                    stack2.pop()
                
            else:
                stack2.append(i)
        return stack1 == stack2

File: Java/#844_backspaceCompare.java
class Solution {
    public boolean backspaceCompare(String S, String T) {
        Stack<Character> s1 = new Stack<Character>();
        Stack<Character> s2 = new Stack<Character>();
        for(int i=0;i<S.length();i++){
            if(S.charAt(i)=='#'){
                if(!s1.isEmpty()) s1.pop();
            }else{
                s1.push(S.charAt(i));
            }
        }
        for(int i=0;i<T.length();i++){
            if(T.charAt(i)=='#'){
                if(!s2.isEmpty()) s2.pop();
            }else{
                s2.push(T.charAt(i));
            }
        }
        
        while(!s1.isEmpty() && !s2.empty()){
            if(s1.pop()!=s2.pop()) return false;
        }
        if(!s1.isEmpty()||!s2.isEmpty()) return false;
        return true;
    }
}
